internal/compose/compose.go

```diff
@@ -12,17 +12,25 @@ import (
 )
 
 const (
-	delimiter    = "\\$"
-	substitution = "[_a-z][._a-z0-9]*(?::?[-?][^}]*)?"
+	delimiter           = "\\$"
+	substitution        = "[_a-z][._a-z0-9-]*"
+	composeSubstitution = "[_a-z][._a-z0-9]*(?::?[-?][^}]*)?"
```

Comment:
Shoud we add the `.` here? The cli doesn't have it, I'm not sure we really need it

Reply:
It's a true question. I added the `.` to make this test pass https://github.com/docker/app/blob/8d52f6d458163e8d5cac6147f6cb4c54092168af/internal/packager/init_test.go#L47
But I'm not sure if the tested usecase is legit

Reply:
I think the test is bogus since env variables can only be alphanumeric and `_`.

Reply:
Dots are necessary for the indentation levels in the parameters `yaml` file. 
As in: https://github.com/docker/app/blob/8d52f6d458163e8d5cac6147f6cb4c54092168af/internal/packager/init_test.go#L75-L78

Reply:
Yeah but here we are talking about [variable substitution](https://docs.docker.com/compose/compose-file/) that the cli does, it can substitute an environment variable inside a compose file (using `${VARIABLE:-default_value}` notation. And an environment variable can only have `[a-zA-Z_]+`. No dots.

Reply:
The test here is verifying the conversion of the valid compose file into a dockerapp file. But the compose file syntax does not support the dot notation.

Reply:
So one could say that it’s not a valid compose file :)